src/Components/RegisterForm.js

```diff
@@ -1,38 +1,80 @@
 import React, { Component } from 'react';
-import {userCreate} from "../APIs/apiService.js";
+import { userCreate } from "../APIs/apiService.js";
 import { userCreation } from "../redux/actions/Authorization";
 import { connect } from 'react-redux';
 import { Redirect } from 'react-router';
-import {NavLink} from 'react-router-dom';
+import { NavLink } from 'react-router-dom';
 
 
 class RegisterForm extends Component {
   constructor(props){
     super(props);
     this.state = {
-      userName: '',
-      email: '',
-      password: '',
-      cPassword: ''
+      userName: {
+        value: '',
+        error: ''
+      },
+      email: {
+        value: '',
+        error: ''
+      },
+      password: {
+        value: '',
+        error: ''
+      },
+      cPassword: {
+        value: '',
+        error: ''
+      }
     }
     this.handleInputChange = this.handleInputChange.bind(this);
     this.createUser = this.createUser.bind(this);
+    this.registerIsValid = this.registerIsValid.bind(this);
+  }
+
+  registerIsValid() {
+    let formIsValid = true;
+    if(!this.state.userName.value){
+      formIsValid = false;
+      this.setState({userName: {error: "Enter user name"}});
+    }
+    if(!this.state.email.value){
+      formIsValid = false;
+      this.setState({email: {error: "Enter email"}});
+    }
+    if(!this.state.password.value){
+      formIsValid = false;
+      this.setState({password: {error: "Enter password"}});
+    }
+    if(!this.state.cPassword.value){
+      formIsValid = false;
+      this.setState({cPassword: {error: "Enter password confirmation"}});
```

Comment:
if state has nested objects, you should setState as:
this.setState({user : {...this.state.user, error: "Enter user name"}})
because in your case you will remove all other propeties in nested object